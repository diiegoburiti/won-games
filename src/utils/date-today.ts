export const dateToday = new Date().toISOString().slice(0, 10)
