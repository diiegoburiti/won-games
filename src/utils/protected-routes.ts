import { getSession } from 'next-auth/client'
import { GetServerSidePropsContext } from 'next'

async function protectedRoutes(context: GetServerSidePropsContext) {
  const session = await getSession(context)

  if (!session) {
    context.res.writeHead(302, {
      Location: `/sign-in?callbackUrl=${context.resolvedUrl}`
    })
    context.res.end()
  }

  return session
}

export default protectedRoutes
