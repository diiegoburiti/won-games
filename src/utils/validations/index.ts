import Joi from 'joi'
import { UsersPermissionsRegisterInput } from 'graphql/generated/globalTypes'

const fieldsValidations = {
  username: Joi.string().min(5).required(),
  email: Joi.string()
    .email({ tlds: { allow: false } })
    .required(),
  password: Joi.string().required(),
  confirm_password: Joi.string()
    .valid(Joi.ref('password'))
    .required()
    .messages({ 'any.only': 'confirm password does not match with password' })
}

export type FieldErrors = {
  [key: string]: string
}

function getFieldErros(objError: Joi.ValidationResult) {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type signInValues = Omit<UsersPermissionsRegisterInput, 'username'>

export function signInValidate(values: signInValues) {
  const { email, password } = fieldsValidations
  const schema = Joi.object({ email, password })
  const fieldErros = schema.validate(values, { abortEarly: false })
  return getFieldErros(fieldErros)
}

export function signUpValidate(values: UsersPermissionsRegisterInput) {
  const schema = Joi.object(fieldsValidations)
  const fieldErros = schema.validate(values, { abortEarly: false })
  return getFieldErros(fieldErros)
}
