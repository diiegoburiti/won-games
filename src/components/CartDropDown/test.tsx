import { screen } from '@testing-library/react'
import { render } from 'utils/test-utils'

import items from 'components/CartList/mock'

import CartDropDown from '.'

describe('<CartDrowDown />', () => {
  it('should render the content correctly', () => {
    const { container } = render(
      <CartDropDown total="R$ 300,00" items={items} />
    )

    expect(screen.getByLabelText(/shopping cart/i)).toBeInTheDocument()
    expect(screen.getByText(`${items.length}`)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render DropDown content with cart items and total', () => {
    render(<CartDropDown items={items} total="R$ 300,00" />)

    expect(screen.getByText('R$ 300,00')).toBeInTheDocument()
    expect(screen.getByText(`${items[0].title}`)).toBeInTheDocument()
  })
})
