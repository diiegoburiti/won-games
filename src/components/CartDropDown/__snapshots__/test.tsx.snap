// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartDrowDown /> should render the content correctly 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c3 {
  color: #FAFAFA;
  width: 2.4rem;
  height: 2.4rem;
  position: relative;
}

.c9 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c11 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c13 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c14 {
  color: #FAFAFA;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

.c17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c17:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c8 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c15 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c16 {
  color: #F231A5;
}

.c2 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
  z-index: 50;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
  z-index: 50;
}

.c6::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c19 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
  background: rgba(0,0,0,0.5);
}

.c1 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c1 .c5,
.c1 .c18 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

.c0 .c7 {
  width: 56rem;
}

@media (min-width:768px) {
  .c9 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c11 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c13 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c15 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          aria-label="Shopping Cart"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="c5 c6"
    >
      <div
        class="c7 c8"
      >
        <div
          class="c9"
        >
          <div
            class="c10"
          >
            <div
              class="c11"
            >
              <img
                alt="Red Dead Redemption 2"
                src="https://source.unsplash.com/user/willianjusten/151x70"
              />
            </div>
            <div
              class="c12"
            >
              <h3
                class="c13"
              >
                Red Dead Redemption 2
              </h3>
              <div
                class="c14"
              >
                R$ 215,00
              </div>
            </div>
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c10"
          >
            <div
              class="c11"
            >
              <img
                alt="Borderlands 3"
                src="https://source.unsplash.com/user/willianjusten/151x70"
              />
            </div>
            <div
              class="c12"
            >
              <h3
                class="c13"
              >
                Borderlands 3
              </h3>
              <div
                class="c14"
              >
                R$ 215,00
              </div>
            </div>
          </div>
        </div>
        <div
          class="c15"
        >
          <span
            class="c16"
          >
            R$ 300,00
          </span>
          <a
            class="c17"
          >
            <span>
              Buy Now
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="c18 c19"
    />
  </div>
</main>
`
